src/Symfony/Component/Lock/Store/RetryTillSaveStore.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * RetryTillSaveStore is a StoreInterface implementation which decorate a non blocking StoreInterface to provide a
+ * kind of blocking storage.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class RetryTillSaveStore implements StoreInterface, ExpirableStoreInterface, BlockingStoreInterface
+{
+    private $decorated;
+    private $retryCount;
+    private $retrySleep;
+
+    /**
+     * @param StoreInterface $decorated  The decorated StoreInterface
+     * @param int            $retrySleep Duration in ms between 2 retry
+     * @param int            $retryCount Maximum amount of retry (0 mean unlimited)
+     */
+    public function __construct(StoreInterface $decorated, $retrySleep = 50, $retryCount = PHP_INT_MAX)
+    {
+        $this->decorated = $decorated;
+        $this->retryCount = $retryCount;
+        $this->retrySleep = $retrySleep;
```

Comment:
The arguments here are assigned in a different sequence than listed in the arguments of the constructor (same as the properties, they are incorrect too). I think it's better to follow the constructor sequence